core/src/main/scala/akka/kafka/internal/BatchingFlowStage.scala

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 - 2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.kafka.internal
+
+import akka.annotation.InternalApi
+import akka.kafka.CommitterSettings
+import akka.kafka.ConsumerMessage.{Committable, CommittableOffsetBatch}
+import akka.kafka.internal.BatchingFlowStage.FlushableOffsetBatch
+import akka.stream._
+import akka.stream.stage._
+
+import scala.util.{Failure, Success, Try}
+
+/**
+ * INTERNAL API.
+ *
+ * Combined stage for producing, batching commits and committing.
+ */
+@InternalApi
+private[kafka] final class BatchingFlowStage(val committerSettings: CommitterSettings)
+    extends GraphStage[FlowShape[Committable, Try[FlushableOffsetBatch]]] {
```

Comment:
Yes , this is a long story. 
On public API level nothing has changed (no Try is exposed) so my expectation/assumption is that it's ok to use `Try`.
I propagate `Try` because otherwise the behaviour was non-deterministic in case of upstream failure. 
Basically sometimes successful batch that comes before the failure also gets 'cancelled'. This triggered random failures in `BatchingFlowStageSpec."batch commit all buffered elements if upstream has suddenly failed"`.  I've identified that this happens due to async step in the `Committer` (replacing asyncMap with map fixes it but we loose parallelism). I have not found a way around this and `Try` seems rather simple solution. Also note that only batch is wrapped in `Try` (not every offset as in the very first version).